### PR TITLE
apps/examples/testcase: Add libc timer testcase

### PR DESCRIPTION
**Pull request by hasw7569**
Add libc timer testcase to improve coverage

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [dfebf85e0a453a84884e8bfde4897edaac30e76f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [dfebf85e0a453a84884e8bfde4897edaac30e76f] - Code Rule Check Result:
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:432: ERROR: [IDT_M_TAB] please, no space before tabs
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:633: ERROR: [SPC_M_OPR] space prohibited after that '*' (ctx:ExW)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:633: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:634: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:635: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:666: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:667: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:668: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:669: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)

**Comment by seinfra** [comment]
Target : [fa6e15c85a031f0a337884370f2be8cb8188f9fe] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [fa6e15c85a031f0a337884370f2be8cb8188f9fe] - Code Rule Check Result:
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:432: ERROR: [IDT_M_TAB] please, no space before tabs
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:633: ERROR: [SPC_M_OPR] space prohibited after that '*' (ctx:ExW)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:633: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:634: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:635: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:666: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:667: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:668: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:669: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)

**Comment by seinfra** [comment]
Target : [0fb9231c19257c9b58010881fbd2223e5986a36f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0fb9231c19257c9b58010881fbd2223e5986a36f] - Code Rule Check Result:
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:432: ERROR: [IDT_M_TAB] please, no space before tabs
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:633: ERROR: [SPC_M_OPR] space prohibited after that '*' (ctx:ExW)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:633: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:634: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:635: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:666: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:667: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:668: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:669: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)

**Comment by seinfra** [comment]
Target : [56fa0652fe9d91b6aecaf6883a21c0134cb9018f] - Code Rule Check Result:
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:432: ERROR: [IDT_M_TAB] please, no space before tabs
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:433: ERROR: [IDT_M_TAB] please, no space before tabs
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:434: ERROR: [IDT_M_TAB] please, no space before tabs
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:435: ERROR: [IDT_M_TAB] please, no space before tabs
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:633: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:634: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:635: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:666: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:667: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:668: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:669: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:WxV)

**Comment by seinfra** [comment]
Target : [56fa0652fe9d91b6aecaf6883a21c0134cb9018f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [04e24df4b9bb8c5787e8b2354de70c4bcde308ea] - Code Rule Check Result:
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:433: ERROR: [IDT_M_TAB] please, no space before tabs
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:434: ERROR: [IDT_M_TAB] please, no space before tabs
apps/examples/testcase/le_tc/kernel/tc_libc_timer.c:435: ERROR: [IDT_M_TAB] please, no space before tabs

**Comment by seinfra** [comment]
Target : [04e24df4b9bb8c5787e8b2354de70c4bcde308ea] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [892c93d2b29d11a7ac5ba2a5b0b7a501d8c8e9f4] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [892c93d2b29d11a7ac5ba2a5b0b7a501d8c8e9f4] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [e8cececc4ed86598626cf13866e0504471280c88] -

**Comment by seinfra** [comment]
Target : [e8cececc4ed86598626cf13866e0504471280c88] -

**Comment by seinfra** [comment]
Target : [01d22de4361ee19f9cbe18a51d2936534e83e96b] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [01d22de4361ee19f9cbe18a51d2936534e83e96b] - Code Rule Check (C++) OK.

File: apps/examples/testcase/le_tc/kernel/tc_libc_timer.c [hunk 1]
```diff
@@ -339,7 +343,7 @@ static void tc_libc_timer_strftime(void)
 	TC_ASSERT_EQ("strftime", strcmp(buffer, "June"), 0);
 
 	/* Check the century number (year/100) as a 2-digit integer */
-	strftime(buffer, BUFF_SIZE, "%C", &st_time);
+	strftime(buffer, BUFF_SIZE, "%y", &st_time);
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/kernel/tc_libc_timer.c]: Please set the testcase sequence by alphabet order.

File: apps/examples/testcase/le_tc/kernel/tc_libc_timer.c [hunk 2]
```diff
@@ -437,6 +450,70 @@ static void tc_libc_timer_strptime(void)
...
+	ret = strptime("Fail", "%U", &sp_tm);
+	TC_ASSERT_EQ("strptime", ret, NULL);
+
+	ret = strptime("Fail", "%Y", &sp_tm);
+	TC_ASSERT_EQ("strptime", ret, NULL);
+
+	ret = strptime("Fail", "%y", &sp_tm);
+	TC_ASSERT_EQ("strptime", ret, NULL);
+
+	memset(&sp_tm, 0, sizeof(struct tm));
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/kernel/tc_libc_timer.c]: Please check that which case needs memset.

File: apps/examples/testcase/le_tc/kernel/tc_libc_timer.c [hunk 3]
```diff
@@ -393,35 +406,35 @@ static void tc_libc_timer_strftime(void)
...
-*                  structure as per the format specifiers.
-* API's covered   :strptime
-* Preconditions   :none
-* Postconditions  :none
-* @return         :void
+* @fn				:tc_libc_timer_strptime
+* @brief			:strptime - Format string as time
+* @Scenario			:strptime - converts time string as the time described in timeptr
+*					 structure as per the format specifiers.
+* API's covered		:strptime
```
Comment (yashwanth686007) [inline apps/examples/testcase/le_tc/kernel/tc_libc_timer.c]: Please check indentation and remove unnecessary gaps.

**Comment by seinfra** [comment]
Target : [3de676f5484ab6ae9e1ccf972bb3b74c44816708] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3de676f5484ab6ae9e1ccf972bb3b74c44816708] - Code Rule Check (C++) OK.

File: apps/examples/testcase/le_tc/kernel/tc_libc_timer.c (hunk 1 above)
Comment (hasw7569) [reply apps/examples/testcase/le_tc/kernel/tc_libc_timer.c]: modify

File: apps/examples/testcase/le_tc/kernel/tc_libc_timer.c (hunk 2 above)
Comment (hasw7569) [reply apps/examples/testcase/le_tc/kernel/tc_libc_timer.c]: modify

File: apps/examples/testcase/le_tc/kernel/tc_libc_timer.c [hunk 4]
```diff
@@ -327,21 +326,20 @@ static void tc_libc_timer_strftime(void)
...
 
-	strftime(buffer, BUFF_SIZE, "%m", &st_time);
-	TC_ASSERT_EQ("strftime", atoi(buffer), st_time.tm_mon + 1);
+	/* Check the zero value */
+	TC_ASSERT_EQ("strftime", strftime(buffer, 0, "return_0_check", &st_time), 0);
 
-	/* Check the abbreviated month name */
-	strftime(buffer, BUFF_SIZE, "%h", &st_time);
-	TC_ASSERT_EQ("strftime", strcmp(buffer, "Jun"), 0);
+	strftime(buffer, BUFF_SIZE, "a", &st_time);
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/kernel/tc_libc_timer.c]: Please add explanation about below tc.

**Comment by seinfra** [comment]
Target : [dc7c011aab59c23401c07f91fdb4ded67d060cfc] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [dc7c011aab59c23401c07f91fdb4ded67d060cfc] - Code Rule Check (C++) OK.

File: apps/examples/testcase/le_tc/kernel/tc_libc_timer.c [hunk 5]
```diff
@@ -327,20 +326,20 @@ static void tc_libc_timer_strftime(void)
...
-	/* Check the abbreviated month name */
-	strftime(buffer, BUFF_SIZE, "%h", &st_time);
-	TC_ASSERT_EQ("strftime", strcmp(buffer, "Jun"), 0);
+	/* Check the three-letter abbreviation for the day of the week */
+	strftime(buffer, BUFF_SIZE, "a", &st_time);
+	TC_ASSERT_EQ("strftime", strncmp(buffer, "a", 2), 0);
+
+	/* Check the full name for the day of the week */
+	strftime(buffer, BUFF_SIZE, "%a", &st_time);
+	TC_ASSERT_EQ("strftime", strncmp(buffer, "Day", 4), 0);
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/kernel/tc_libc_timer.c]: Please change strncmp size to calculated number with strlen

**Comment by seinfra** [comment]
Target : [6787e781d6846c06a87806298788f62c00914d53] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [6787e781d6846c06a87806298788f62c00914d53] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [eb19a9562ed14f6d650532263a4567f38a63d181] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [eb19a9562ed14f6d650532263a4567f38a63d181] - Code Rule Check (C++) OK.

File: apps/examples/testcase/le_tc/kernel/tc_libc_timer.c (hunk 3 above)
Comment (hasw7569) [reply apps/examples/testcase/le_tc/kernel/tc_libc_timer.c]: modify

File: apps/examples/testcase/le_tc/kernel/tc_libc_timer.c (hunk 4 above)
Comment (hasw7569) [reply apps/examples/testcase/le_tc/kernel/tc_libc_timer.c]: modify

File: apps/examples/testcase/le_tc/kernel/tc_libc_timer.c (hunk 5 above)
Comment (hasw7569) [reply apps/examples/testcase/le_tc/kernel/tc_libc_timer.c]: modify

**Comment by seinfra** [comment]
Target : [45d55d90d42d4120632e5a867bf5c13a5c924e1b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [45d55d90d42d4120632e5a867bf5c13a5c924e1b] - Code Rule Check OK.

**Comment by sunghan-chang** [comment]
@hasw7569 Could you add test logs in commit description?

**Comment by seinfra** [comment]
Target : [82788178effaab78eea75a51703b4d19ea1174e3] -

**Comment by seinfra** [comment]
Target : [82788178effaab78eea75a51703b4d19ea1174e3] -

**Pull request merged**